tests/sai_qualify/sai_infra.py

```diff
@@ -76,10 +79,42 @@ def sai_test_env_check(creds, duthost, ptfhost, request):
         logger.info("Test env check is failed in previous check. \
             Fails this check directly.")
         raise Exception("SAI Test env error.")
-
     check_test_env_with_retry(creds, duthost, ptfhost, request)
 
 
+@pytest.fixture(scope="module")
+def start_warm_reboot_watcher(creds, duthost, request, ptfhost, create_sai_test_interface_param):
+    # install apscheduler before running
+    logger.info("start warm reboot watcher")
+    logging.getLogger("apscheduler.executors.default").propagate = False
+    logging.getLogger("apscheduler.executors.default").setLevel(logging.ERROR)
+    ptfhost.shell("touch {}".format("/tmp/warm_reboot"))
+    ptfhost.shell("echo  > {}".format("/tmp/warm_reboot"))
+    scheduler = BackgroundScheduler(
+        {'apscheduler.job_defaults.max_instances': 1})
+
+    scheduler.add_job(warm_reboot_change_handler, "cron", [
+                      creds, duthost, request, ptfhost, create_sai_test_interface_param], second="*/1")
+    scheduler.start()
+
+
+def warm_reboot_change_handler(creds, duthost, request, ptfhost, create_sai_test_interface_param):
```

Comment:
this is a critial method.
please add log message for each step in this method.
if file content changed, pleae also log that information.